rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictFluentReturnRector/Fixture/inner_classmethod.php.inc

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictFluentReturnRector\Fixture;
+
+final class InnerClassMethod
+{
+    public function test()
+    {
+        new class {
+            public function run()
+            {
+                return $this;
```

Comment:
Do we need support for ` return new self()` or `return new static();` ?

Reply:
I think that need separate PR for easier to review ;)